File: src/main/scala/org/xbery/artbeams/comments/repository/CommentFilter.scala
package org.xbery.artbeams.comments.repository

import org.xbery.artbeams.common.assets.repository.AssetFilter

/**
  * @author Radek Beran
  */
case class CommentFilter(
  override val id: Option[String],
  override val ids: Option[Seq[String]],
  override val createdBy: Option[String],
  entityId: Option[String]
) extends AssetFilter

object CommentFilter {
  lazy val Empty = CommentFilter(None, None, None, None)

  def forEntityId(entityId: String) = CommentFilter.Empty.copy(entityId = Some(entityId))
}

File: src/main/kotlin/org/xbery/artbeams/comments/repository/CommentFilter.kt
package org.xbery.artbeams.comments.repository

import org.xbery.artbeams.common.assets.repository.AssetFilter

/**
 * @author Radek Beran
 */
data class CommentFilter(override val id: String?, override val ids: List<String>?, override val createdBy: String?, val entityId: String?) : AssetFilter {
  companion object  {
     val Empty: CommentFilter = CommentFilter(null, null, null, null)
    fun forEntityId(entityId: String): CommentFilter = CommentFilter.Empty.copy(entityId = entityId)
  }
}
